src/cr/cube/dimension.py

```diff
@@ -525,6 +559,135 @@ def valid_elements(self):
+          the index integer here. Long term zz9 may change this to the the alias.
+        - "subvariable_id": Subvariables have an id stored in
+          `dimensions[i].type.elements[j].value.id`, generally this is a 4 digit,
+          0-padded index of the subvariable when it was first created (eg "0001",
+          "0002", ...), though it is not required to be. For derived insertions,
+          currently the name is used here.
+        - "alias": Subvariables also have an alias that identifies them. It is stored
+          in `dimensions[i].type.elements[j].value.references.alias`.
+        """
+        shim = copy.deepcopy(self._dimension_transforms_dict)
+
+        # --- Leave non-subvariable dimension types alone, as they don't have
+        # --- subvariable aliases to use, and category ids are already the main way
+        # --- we identify elements on categorical dimensions (and this is correct).
+        if self._dimension_type not in DT.ARRAY_TYPES:
+            return shim
+
+        # --- Replace element transform ids with the alias
+        # --- The name of the element transform object is a string. This string is
+        # --- assumed to refer to the subvariable id, unless there is a "key", which
+        # --- can be "alias", in which case it refers to the alias, or if no
+        # --- subvariable is found with the id, in which case it's assumed to be the
+        # --- element_id
+        element_transforms = self._dimension_transforms_dict.get("elements", {})
+        # --- TODO: This logic about "key" is not supported by the validation
+        # --- on the deck schema, so maybe we should remove it?
+        key = element_transforms.get("key")
+        if key == "subvar_id":
+            # --- translate from subvariable id
+            old_keys = tuple(element_transforms.keys())
+            new_keys = tuple(
+                self._subvar_aliases[self._subvar_ids.index(_id)]
+                for _id in old_keys
+                if _id in self._subvar_ids
+            )
+            shim["elements"] = {
+                nkey: element_transforms[old_keys[i]] for i, nkey in enumerate(new_keys)
+            }
+        elif key != "alias":
+            old_keys = tuple(element_transforms.keys())
+            new_keys = tuple(self.translate_element_id(_id) for _id in old_keys)
+            shim["elements"] = {
+                nkey: element_transforms[old_keys[i]]
+                for i, nkey in enumerate(new_keys)
+                if nkey is not None
+            }
+
+        # --- Translate explicit order element ids if present
+        element_ids = self._dimension_transforms_dict.get("order", {}).get(
+            "element_ids"
+        )
+        if element_ids:
+            shim["order"]["element_ids"] = [
+                self.translate_element_id(_id) for _id in element_ids
+            ]
+
+        # --- sort-by-value on the opposing dimension also refers to element ids, but
```

Comment:
Not sure if this is the right approach, but having to look up opposing dimension here seemed difficult